<?xml version="1.0" encoding="utf-8"?>
<launch>

    <!-- Load default extrinsics. NOTE: Adding 20cm to the height of the camera produces better results. At least on visual inspection -->
    <include file="$(find segmentation_filtered_icp)/launch/versavis/extrinsics.launch"/>

    <!-- Start SMB State estimator -->
    <include file="$(find smb_state_estimator)/launch/smb_state_estimator_standalone.launch"/>

    <!-- Start Selective ICP -->
    <include file="$(find cpt_selective_icp)/launch/supermegabot_selective_icp.launch" pass_all_args="true">
        <arg name="publish_distance" value="true"/>
    </include>

    <!-- Start Dataset Creator node for each camera -->
    <node pkg="background_foreground_segmentation" type="background_foreground_segmentation_node"
          name="dataset_creator_cam0" output="screen">
        <rosparam command="load" file="$(dirname)/../config/cla_dataset_creator_in_memory.yaml"/>
        <param name="outputFolder" value="/tmp/"/>


        <param name="cameraImageTopic" value="/camera_stick/cam0/image"/>
        <param name="cameraInfoTopic" value="/camera_stick/cam0/camera_info"/>
        <param name="cameraFrame" value="/cam0"/>

        <param name="outputTopic" value="/meshdist/cam0/original"/>
        <param name="labelsTopic" value="/meshdist/cam0/projected_pc"/>
        <param name="distanceTopic" value="/meshdist/cam0/projected_pc_distance"/>


    </node>

    <node pkg="background_foreground_segmentation" type="background_foreground_segmentation_node"
          name="dataset_creator_cam1" output="screen">

        <rosparam command="load" file="$(dirname)/../config/cla_dataset_creator_in_memory.yaml"/>
        <param name="outputFolder" value="/tmp/"/>
        <param name="cameraImageTopic" value="/camera_stick/cam1/image"/>
        <param name="cameraInfoTopic" value="/camera_stick/cam1/camera_info"/>
        <param name="cameraFrame" value="/cam1"/>

        <param name="outputTopic" value="/meshdist/cam1/original"/>
        <param name="labelsTopic" value="/meshdist/cam1/projected_pc"/>
        <param name="distanceTopic" value="/meshdist/cam1/projected_pc_distance"/>
    </node>

    <node pkg="background_foreground_segmentation" type="background_foreground_segmentation_node"
          name="dataset_creator_cam2" output="screen">
        <rosparam command="load" file="$(dirname)/../config/cla_dataset_creator_in_memory.yaml"/>
        <param name="outputFolder" value="/tmp/"/>


        <param name="cameraImageTopic" value="/camera_stick/cam2/image"/>
        <param name="cameraInfoTopic" value="/camera_stick/cam2/camera_info"/>
        <param name="cameraFrame" value="/cam2"/>

        <param name="outputTopic" value="/meshdist/cam2/original"/>
        <param name="labelsTopic" value="/meshdist/cam2/projected_pc"/>
        <param name="distanceTopic" value="/meshdist/cam2/projected_pc_distance"/>
    </node>

    <!-- Start RVIZ -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find cpt_selective_icp)/launch/rviz.rviz"/>


</launch>

